src/main/drivers/max7456.c

```diff
@@ -536,10 +535,11 @@ void max7456WriteChar(uint8_t x, uint8_t y, uint8_t c)
 
 void max7456Write(uint8_t x, uint8_t y, const char *buff)
 {
-    uint8_t i = 0;
-    for (i = 0; *(buff+i); i++)
-        if (x+i < CHARS_PER_LINE) // Do not write over screen
+    for (int i = 0; *(buff+i); i++) {
```

Comment:
```
for (int i = 0; buff[i]; i++) {
...
screenBuffer[y*CHARS_PER_LINE+x+i] = buff[i];
```
Or even better:
```
int cpy = MIN(strlen(buff), CHARS_PER_LINE - x);  // stop at end of line
if(cpy>0) {
  memcpy(screenBuffer + y * CHARS_PER_LINE + x, buff, cpy);
}

Reply:
This change adds 16 bytes to code size.

Reply:
Well, the strlen call is surprisingly expensive - it has to stash 3 registers before call.
16 bytes may be caused by alignment, actual size increase MAY be lower ... 

This will generate tighter code, but will pull in `strnlen`, so it will be probably longer :
```
void max7456Write(uint8_t x, uint8_t y, const char *buff)
{
    void* dst = screenBuffer + y * CHARS_PER_LINE + x;
    memcpy(dst, buff, strnlen(buff, CHARS_PER_LINE - x));
}
```